### ISSUE TITLE
`same_file::is_same_file` does work to identify assembly file on MacOS

### ISSUE THREAD

**Issue by konkers** [pacak/cargo-show-asm#76]
The code which locates the assembly file for an executable target uses `same_file::is_same_file` to find the file in `deps/` that is that same as the target binary.  On Linux this works because cargo hard links the two (see the inode matching in the first column of the `ls -li` output:

```bash
linux-box> ls -li target/thumbv6m-none-eabi/release/rp2040-project-template
43257897 -rwx------ 2 konkers primarygroup 462184 Oct 26 23:19 target/thumbv6m-none-eabi/release/rp2040-project-template
linux-box> ls -li target/thumbv6m-none-eabi/release/deps/rp2040_project_template-ada3da3d6c134605
43257897 -rwx------ 2 konkers primarygroup 462184 Oct 26 23:19 target/thumbv6m-none-eabi/release/deps/rp2040_project_template-ada3da3d6c134605
```

However, on MacOS, it appears that cargo is copying the file as they have different inodes:
```bash
mac-box> ls -li target/thumbv6m-none-eabi/release/deps/rp2040_project_template-230a92e91f598f68
64623547 -rwx------  1 konkers  primarygroup  433724 Oct 28 09:56 target/thumbv6m-none-eabi/release/deps/rp2040_project_template-230a92e91f598f68
mac-box> ls -li target/thumbv6m-none-eabi/release/rp2040-project-template
64629801 -rwx------  1 konkers  primarygroup  433724 Oct 28 09:56 target/thumbv6m-none-eabi/release/rp2040-project-template
```

Perhaps comparing the actual contents is needed on MacOS?

**Comment by konkers** [comment]
Here is a naive fix for the issue: https://github.com/konkers/cargo-show-asm/commit/cbc579401859414c413e7cb9f2e55253f8a0c5d2

**Comment by pacak** [comment]
Thank you for the report, I was hoping MacOS is close enough to Linux so it would just work.

I pushed a branch `mac` it seems to be doing the right thing detecting the file but the asm it prints feels a bit odd. Is it displaying the right asm code for you? Was it working correctly in pre 0.2.0 version?

**Comment by pacak** [comment]
Also which mac is it? Intel or newer arm based?

**Comment by pacak** [comment]
Anyway, fixed in master. Can you check if it works for you? If so - I'll release.

**Comment by pacak** [comment]
Published 0.2.1, the bug should be fixed along one more Mac specific issue, please reopen the ticket if the problem persists.

**Comment by konkers** [comment]
Thanks for the quick response!  This is an 2019 Intel MacBook Pro.  I'll test the new release and let you know if this issue persists.
